terminate called after throwing an instance of 'std::runtime_error'
  what():  required args: learning-rate needs a value
